onnxruntime/core/framework/cloud_invoker.cc

```diff
@@ -0,0 +1,290 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#ifdef USE_CLOUD
+#include "http_client.h"
+#include "core/common/common.h"
+#include "core/framework/cloud_invoker.h"
+#include "core/framework/ort_value.h"
+
+#define CHECK_TRITON_ERR(ret, msg)                                                           \
+  if (!ret.IsOk()) {                                                                         \
+    return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, msg, ", triton err: ", ret.Message().c_str()); \
+  }
+
+namespace onnxruntime {
+
+namespace tc = triton::client;
+
+constexpr char* kCloudUri = "cloud.uri";
+constexpr char* kCloudModelName = "cloud.model_name";
+constexpr char* kCloudModelVer = "cloud.model_version";
+constexpr char* kCloudVerbose = "cloud.verbose";
+constexpr char* kCloudEndpointType = "cloud.endpoint_type";
+constexpr char* kCloudAuthKey = "cloud.auth_key";
+constexpr char* kCloudTriton = "triton";
+
+class TritonInvoker : public CloudEndPointInvoker {
+ public:
+  TritonInvoker(const CloudEndPointConfig& config);
+  onnxruntime::Status Send(const CloudEndPointConfig& run_options,
+                           const InlinedVector<std::string>& input_names,
+                           gsl::span<const OrtValue> ort_inputs,
+                           const InlinedVector<std::string>& output_names,
+                           std::vector<OrtValue>& ort_outputs) const noexcept override;
+
+ private:
+  static std::string MapDataType(int32_t ort_data_type);
```

Comment:
Why static? Is there a usage of it outside this class?

Reply:
because the function is a class-level utility.

Reply:
What's the point of making it private then? No one else outside can use it. Not following.

Reply:
Why would anyone want to use the function outside of the class? It is for triton exclusively.